searx/network/client.py

```diff
@@ -192,6 +191,11 @@ def new_client(enable_http, verify, enable_http2,
     return httpx.AsyncClient(transport=transport, mounts=mounts, max_redirects=max_redirects)
 
 
+def get_loop():
+    global LOOP
```

Comment:
There is no need `global LOOP` since LOOP is not assigned (?)

Reply:
> There is no need global LOOP since LOOP is not assigned (?)

To access a global variable inside a function there is no need to use global keyword in general .. but I think it is good practice to use global keyword to explicit define a variable as _global_ instead of being [implicit](https://docs.python.org/3/faq/programming.html#what-are-the-rules-for-local-and-global-variables-in-python).

BTW: LOOP is assigned by `None` in line 25  / otherwise you would get an `NameError: name 'LOOP' is not defined`:

https://github.com/searxng/searxng/blob/283ae7bfad05fedbf6c7a7e7d7addfc3ae7ed9c7/searx/network/client.py#L25

Reply:
>  I think it is good practice to use global keyword to explicit define a variable as global instead of being implicit. 

Ok.
But new code will have a lot of `global` since there are a lot of global variables.

Reply:
[On one hand, requiring global for assigned variables provides a bar against unintended side-effects. On the other hand, if global was required for all global references, you’d be using global all the time. You’d have to declare as global every reference to a built-in function or to a component of an imported module. This clutter would defeat the usefulness of the global declaration for identifying side-effects.](https://docs.python.org/3/faq/programming.html#what-are-the-rules-for-local-and-global-variables-in-python)

> But new code will have a lot of global since there are a lot of global variables.

Hm ... if it is to much, then we can switch to use implicit globals .. lets see ..I can't judge which is better for searx / I do use globals very rare in my projects.